app/screens/buyers/BuyingItemList.java

```diff
@@ -43,35 +44,46 @@ public BuyingItemList(Store store, CATEGORY_ENUM category) {
         } catch (IOException io) {
             io.printStackTrace();
         }
-
     }
 
     public void display() {
+        Item item;
+
         while (true) {
-            int choice;
             MenuTitle.displayStoreName(storeName);
             MenuTitle.displaySubTitle(categoryName);
             MenuTitle.displayOptions(itemNames);
-
-            System.out.print("\nChoice: ");
             try {
-                choice = Integer.parseInt(sc.nextLine());
+                getChoice(); // gets and validates the choice given by the user
                 if (choice == itemList.size()) {
                     return;
                 }
-                Item item = itemList.get(choice - 1);
+                item = itemList.get(choice - 1);
                 this.buyItem(item);
             } catch (UnauthorizedException | CategoryDoesNotExistException | BuyException e) {
                 System.out.println(e.getMessage());
             } catch (IOException e) {
                 e.printStackTrace();
-            } catch (Exception e) {
+            }
+        }
+    }
+
+    private void getChoice() {
```

Comment:
better nga mag return nalang siya sa choice nga int rather than modifying ang instance variable

mas mo fit din siya nga mag return siya ng value kay `get` ang verb